api/v1/helpers/verbs/doFind.js

```diff
@@ -32,19 +34,21 @@ const httpStatus = require('../../constants').httpStatus;
  *  find command
  */
 function doFindAndCountAll(reqResNext, props, opts) {
-  u.getScopedModel(props, opts.attributes).findAndCountAll(opts)
-  .then((o) => {
-    reqResNext.res.set(COUNT_HEADER_NAME, o.count);
-    const retval = o.rows.map((row) => {
-      if (props.modelName === 'Lens') {
-        delete row.dataValues.library;
-      }
+  return new Promise((resolve, reject) => {
+    u.getScopedModel(props, opts.attributes).findAndCountAll(opts)
+    .then((o) => {
+      reqResNext.res.set(COUNT_HEADER_NAME, o.count);
+      const retval = o.rows.map((row) => {
+        if (props.modelName === 'Lens') {
+          delete row.dataValues.library;
+        }
 
-      return u.responsify(row, props, reqResNext.req.method);
-    });
-    reqResNext.res.status(httpStatus.OK).json(retval);
-  })
-  .catch((err) => u.handleError(reqResNext.next, err, props.modelName));
+        return u.responsify(row, props, reqResNext.req.method);
+      });
+      resolve(retval);
+    })
+    .catch((err) => reject(err));
+  });
 } // doFindAndCountAll
```

Comment:
Not sure if there is a better way but what about returning u.getScopedModel? This would return a promise and you can do a "then" in the caller to get the value.